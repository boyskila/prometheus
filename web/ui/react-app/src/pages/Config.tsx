import React, { FC } from 'react';
import { RouteComponentProps } from '@reach/router';

const Config: FC<RouteComponentProps> = () => <div>Config page</div>

export default Config;
