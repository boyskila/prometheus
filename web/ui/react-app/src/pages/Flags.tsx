import React, { FC } from 'react';
import { RouteComponentProps } from '@reach/router';
import { Table } from 'reactstrap';
import { Fetch, FetchState } from '../api/Fetch';
import PathPrefixProps from '../PathPrefixProps';
import { StatusIndicator } from '../StatusIndicator';

export interface FlagMap {
  [key: string]: string;
}

export const FlagsContent: FC<FetchState<FlagMap>> = ({ data = {} }) => {
  return (
    <>
      <h2>Command-Line Flags</h2>
      <Table bordered size="sm" striped>
        <tbody>
          {Object.keys(data).map(key => (
            <tr key={key}>
              <th>{key}</th>
              <td>{data[key]}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </>
  );
};

const Flags: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix = '' }) => {
  return (
    <Fetch url={`${pathPrefix}/api/v1/status/flags`}>
      {({ error, data }: FetchState<FlagMap>) => (
        <StatusIndicator error={error && `Error fetching flags: ${error.message}`} hasData={Boolean(data)}>
          <FlagsContent data={data} />
        </StatusIndicator>
      )}
    </Fetch>
  );
};

export default Flags;
