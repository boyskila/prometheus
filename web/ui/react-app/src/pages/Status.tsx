import React, { Fragment, FC } from 'react';
import { RouteComponentProps } from '@reach/router';
import { Table } from 'reactstrap';
import { FetchState } from '../api/Fetch';
import { fetchWithStatus } from '../api/FetchWithStatus';

const endpoints = ['/api/v1/status/runtimeinfo', '/api/v1/status/buildinfo', '/api/v1/alertmanagers'];
const sectionTitles = ['Runtime Information', 'Build Information', 'Alertmanagers'];

interface StatusConfig {
  [k: string]: { title?: string; customizeValue?: (v: any) => any; customRow?: boolean; skip?: boolean };
}

type StatusPageState = Array<{ [k: string]: string }>;

export const statusConfig: StatusConfig = {
  startTime: { title: 'Start time', customizeValue: (v: string) => new Date(v).toUTCString() },
  CWD: { title: 'Working directory' },
  reloadConfigSuccess: {
    title: 'Configuration reload',
    customizeValue: (v: boolean) => (v ? 'Successful' : 'Unsuccessful'),
  },
  lastConfigTime: { title: 'Last successful configuration reload' },
  chunkCount: { title: 'Head chunks' },
  timeSeriesCount: { title: 'Head time series' },
  corruptionCount: { title: 'WAL corruptions' },
  goroutineCount: { title: 'Goroutines' },
  storageRetention: { title: 'Storage retention' },
  activeAlertmanagers: {
    customRow: true,
    customizeValue: (alertMgrs: { url: string }[]) => {
      return (
        <Fragment key="alert-managers">
          <tr>
            <th>Endpoint</th>
          </tr>
          {alertMgrs.map(({ url }) => {
            const { origin, pathname } = new URL(url);
            return (
              <tr key={url}>
                <td>
                  <a href={url}>{origin}</a>
                  {pathname}
                </td>
              </tr>
            );
          })}
        </Fragment>
      );
    },
  },
  droppedAlertmanagers: { skip: true },
};

export const StatusContent: FC<FetchState<StatusPageState>> = ({ data = [] }) => {
  return (
    <>
      {data.map((statuses, i) => {
        return (
          <Fragment key={i}>
            <h2>{sectionTitles[i]}</h2>
            <Table className="h-auto" size="sm" bordered striped>
              <tbody>
                {Object.entries(statuses).map(([k, v]) => {
                  const { title = k, customizeValue = (val: any) => val, customRow, skip } = statusConfig[k] || {};
                  if (skip) {
                    return null;
                  }
                  if (customRow) {
                    return customizeValue(v);
                  }
                  return (
                    <tr key={k}>
                      <th className="capitalize-title" style={{ width: '35%' }}>
                        {title}
                      </th>
                      <td className="text-break">{customizeValue(v)}</td>
                    </tr>
                  );
                })}
              </tbody>
            </Table>
          </Fragment>
        );
      })}
    </>
  );
};

StatusContent.displayName = 'Status';

export default fetchWithStatus<RouteComponentProps, StatusPageState>(StatusContent, undefined, endpoints);
