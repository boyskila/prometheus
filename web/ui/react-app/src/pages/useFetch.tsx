import React, { useState, useEffect } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';

const useFetch = <R extends {}>(urls: string[], options?: any) => {
  const [response, setResponse] = useState<R[]>();
  const [error, setError] = useState<Error>();
  const [isLoading, setIsLoading] = useState<boolean>();

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const responses = await Promise.all(
          urls.map(async url => {
            const response = await fetch(url)
            if (response.ok) {
              const result = await response.json()
              return result.data
            }
          })
        );
        setResponse(responses);
        setIsLoading(false);
      } catch (error) {
        setError(error);
      }
    };
    fetchData();
  }, [urls]);

  return {
    response,
    error,
    loading: isLoading ? {
      spiner: (
        <FontAwesomeIcon
          size="3x"
          icon={faSpinner}
          spin
          className="position-absolute"
          style={{ transform: 'translate(-50%, -50%)', top: '50%', left: '50%' }}
        />
      ),
    } : null,
  };
};

export default useFetch;
