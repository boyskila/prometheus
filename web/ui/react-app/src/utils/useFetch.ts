import { useState, useEffect } from 'react';

export type APIResponse<T> = { status?: string; data?: T };

export interface FetchState<T> {
  data?: T;
  status?: string;
  error?: Error;
  isLoading: boolean;
}

const initialResponseState: APIResponse<any> = { status: undefined, data: undefined };

export const useFetch = <T extends {}>(url: string, options?: RequestInit): FetchState<T> => {
  const [response, setResponse] = useState<APIResponse<T>>(initialResponseState);
  const [error, setError] = useState<Error>();
  const [isLoading, setIsLoading] = useState<boolean>(false);

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const res = await fetch(url, options);
        if (!res.ok) {
          throw new Error(res.statusText);
        }
        const json = (await res.json()) as APIResponse<T>;
        setResponse(json || initialResponseState);
        setIsLoading(false);
      } catch (error) {
        setError(error);
      }
    };
    fetchData();
  }, [url, options]);
  return { ...response, error, isLoading };
};
