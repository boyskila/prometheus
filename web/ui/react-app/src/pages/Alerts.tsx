import React, { FC, useState } from 'react';
import { RouteComponentProps } from '@reach/router';
import PathPrefixProps from '../PathPrefixProps';
// import { Alert, TabContent, TabPane } from 'reactstrap';
import { useFetch } from '../utils/useFetch';
import { withStatusIndicator } from '../withStatusIndicator';
import { ButtonGroup, Alert, Collapse, Card } from 'reactstrap';
// import { Button, Collapse } from 'reactstrap';

interface AlertsProps {
  groups?: any[];
}

const byAlertType = (group: any) => group.rules.some((e: any) => e.type === 'alerting');

export const AlertsContent: FC<AlertsProps> = ({ groups = [] }) => {
  const alertGroups = groups.filter(byAlertType);
  console.log(alertGroups);

  return <ButtonGroup>{Object}</ButtonGroup>;
};
const AlertsWithStatusIndicator = withStatusIndicator(AlertsContent);

AlertsContent.displayName = 'Alerts';

const Colap = ({ rule }: any) => {
  const [open, toggle] = useState(false)
  console.log(rule);

  return (
    <>
      <Alert onClick={() => toggle(!open)} color="danger">{rule.name}({`${rule.alerts.length} active`})</Alert>
      <Collapse isOpen={open}>
        <Card>
          <div>name: {rule.name}</div>
          <div>expr: {rule.query}</div>
          <div>
            <div>labels:</div>
            <div className="ml-5">severity: {rule.labels.severity}</div>
          </div>
          <div>
            <div>annotations:</div>
            <div className="ml-5">summary: {rule.annotations.summary}</div>
          </div>
        </Card>
      </Collapse>
    </>
  )
}

const Alerts: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix = '' }) => {
  // const { response, error, isLoading } = useFetch<AlertsProps>(`${pathPrefix}/api/v1/rules`);
  const { response } = useFetch<AlertsProps>(`${pathPrefix}/api/v1/rules?type=alert`);
  response.data && console.log(response.data.groups);
  // return <AlertsWithStatusIndicator groups={response.data && response.data.groups} error={error} isLoading={isLoading} />;
  if (response.data && response.data.groups) {
    return (
      <div>
        {response.data.groups.map(g => {
          return (
            <>
              <div>{g.file} > {g.name}</div>
              {g.rules.map((rule: any) => {
                return <Colap rule={rule} />
              })}
            </>
          )
        })}
      </div>
    );
  }
  return null
};

export default Alerts;
