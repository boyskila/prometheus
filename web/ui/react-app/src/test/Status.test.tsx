import * as React from 'react';
import { shallow } from 'enzyme';
import { Status } from '../pages';
import { Alert, Table } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as useFetch from '../pages/useFetch';
import renderer from 'react-test-renderer';

describe('Status', () => {
  afterEach(() => jest.restoreAllMocks());

  it('should render spinner while waiting data', () => {
    const wrapper = shallow(<Status />);
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(1);
  });
  it('should render Alert on error', () => {
    jest.spyOn(useFetch, 'default').mockImplementation(() => ({ error: new Error('foo') } as any));
    const wrapper = shallow(<Status />);
    expect(wrapper.find(Alert)).toHaveLength(1);
  });
  it('should fetch proper API endpoints', () => {
    const useFetchSpy = jest.spyOn(useFetch, 'default');
    shallow(<Status />);
    expect(useFetchSpy).toHaveBeenCalledWith(['../api/v1/runtimeinfo', '../api/v1/buildinfo', '../api/v1/alertmanagers']);
  });
  describe('Status Table', () => {
    const response = [
      {
        StartTime: '2019-10-30T22:03:23.247913868+02:00',
        CWD: '/home/boyskila/Desktop/prometheus',
        GoroutineCount: 37,
        GOMAXPROCS: 4,
        GOGC: '',
        GODEBUG: '',
        CorruptionCount: 0,
        ChunkCount: 1383,
        TimeSeriesCount: 461,
        LastConfigTime: '2019-10-30T22:03:23+02:00',
        ReloadConfigSuccess: true,
        StorageRetention: '15d',
      },
      {
        version: '',
        revision: '',
        branch: '',
        buildUser: '',
        buildDate: '',
        goVersion: 'go1.13.3',
      },
      {
        activeAlertmanagers: [
          { url: 'https://1.2.3.4:9093/api/v1/alerts' },
          { url: 'https://1.2.3.5:9093/api/v1/alerts' },
          { url: 'https://1.2.3.6:9093/api/v1/alerts' },
          { url: 'https://1.2.3.7:9093/api/v1/alerts' },
          { url: 'https://1.2.3.8:9093/api/v1/alerts' },
          { url: 'https://1.2.3.9:9093/api/v1/alerts' },
        ],
        droppedAlertmanagers: [],
      },
    ];

    afterEach(() => jest.restoreAllMocks());

    it('should display 3 tables', () => {
      jest.spyOn(useFetch, 'default').mockImplementation(() => ({ response } as any));
      const wrapper = shallow(<Status />);
      expect(wrapper.find(Table)).toHaveLength(3);
    });
    it('should titles have proper positions', () => {
      jest.spyOn(useFetch, 'default').mockImplementation(() => ({ response } as any));
      const wrapper = shallow(<Status />);
      expect(
        wrapper
          .find('h2')
          .at(0)
          .text()
      ).toEqual('Runtime Information');
      expect(
        wrapper
          .find('h2')
          .at(1)
          .text()
      ).toEqual('Build Information');
      expect(
        wrapper
          .find('h2')
          .at(2)
          .text()
      ).toEqual('Alertmanagers');
    });
    it('should match table snapshot', () => {
      jest.spyOn(useFetch, 'default').mockImplementation(() => ({ response } as any));
      const wrapper = renderer.create(<Status />);
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
