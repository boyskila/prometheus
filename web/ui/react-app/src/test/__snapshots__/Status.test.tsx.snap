// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Status Table Data should be placed properly within the table 1`] = `
Array [
  <h2>
    Runtime Information
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Start time
          </td>
          <td
            className="m-0 row"
          >
            Wed, 30 Oct 2019 20:03:23 GMT
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Working directory
          </td>
          <td
            className="m-0 row"
          >
            /home/boyskila/Desktop/prometheus
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Goroutines
          </td>
          <td
            className="m-0 row"
          >
            37
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GOMAXPROCS
          </td>
          <td
            className="m-0 row"
          >
            4
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GOGC
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GODEBUG
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            WAL corruptions
          </td>
          <td
            className="m-0 row"
          >
            0
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Head chunks
          </td>
          <td
            className="m-0 row"
          >
            1383
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Head time series
          </td>
          <td
            className="m-0 row"
          >
            461
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Last successful configuration reload
          </td>
          <td
            className="m-0 row"
          >
            2019-10-30T22:03:23+02:00
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Configuration reload
          </td>
          <td
            className="m-0 row"
          >
            Successful
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Storage retention
          </td>
          <td
            className="m-0 row"
          >
            15d
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <h2>
    Build Information
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            version
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            revision
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            branch
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            buildUser
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            buildDate
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            goVersion
          </td>
          <td
            className="m-0 row"
          >
            go1.13.3
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <h2>
    Alertmanagers
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Active
          </td>
          <td
            className="m-0 row"
          >
            <a
              className="mr-5"
              href="https://1.2.3.4:9093/api/v1/alerts"
            >
              https://1.2.3.4:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.5:9093/api/v1/alerts"
            >
              https://1.2.3.5:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.6:9093/api/v1/alerts"
            >
              https://1.2.3.6:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.7:9093/api/v1/alerts"
            >
              https://1.2.3.7:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.8:9093/api/v1/alerts"
            >
              https://1.2.3.8:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.9:9093/api/v1/alerts"
            >
              https://1.2.3.9:9093/api/v1/alerts
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Dropped
          </td>
          <td
            className="m-0 row"
          />
        </tr>
      </tbody>
    </table>
  </div>,
]
`;

exports[`Status Status Table Data should be placed properly within the table should render Runtime Information table 1`] = `
Array [
  <h2>
    Runtime Information
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Start time
          </td>
          <td
            className="m-0 row"
          >
            Wed, 30 Oct 2019 20:03:23 GMT
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Working directory
          </td>
          <td
            className="m-0 row"
          >
            /home/boyskila/Desktop/prometheus
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Goroutines
          </td>
          <td
            className="m-0 row"
          >
            37
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GOMAXPROCS
          </td>
          <td
            className="m-0 row"
          >
            4
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GOGC
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GODEBUG
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            WAL corruptions
          </td>
          <td
            className="m-0 row"
          >
            0
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Head chunks
          </td>
          <td
            className="m-0 row"
          >
            1383
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Head time series
          </td>
          <td
            className="m-0 row"
          >
            461
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Last successful configuration reload
          </td>
          <td
            className="m-0 row"
          >
            2019-10-30T22:03:23+02:00
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Configuration reload
          </td>
          <td
            className="m-0 row"
          >
            Successful
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Storage retention
          </td>
          <td
            className="m-0 row"
          >
            15d
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <h2>
    Build Information
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            version
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            revision
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            branch
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            buildUser
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            buildDate
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            goVersion
          </td>
          <td
            className="m-0 row"
          >
            go1.13.3
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <h2>
    Alertmanagers
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Active
          </td>
          <td
            className="m-0 row"
          >
            <a
              className="mr-5"
              href="https://1.2.3.4:9093/api/v1/alerts"
            >
              https://1.2.3.4:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.5:9093/api/v1/alerts"
            >
              https://1.2.3.5:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.6:9093/api/v1/alerts"
            >
              https://1.2.3.6:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.7:9093/api/v1/alerts"
            >
              https://1.2.3.7:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.8:9093/api/v1/alerts"
            >
              https://1.2.3.8:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.9:9093/api/v1/alerts"
            >
              https://1.2.3.9:9093/api/v1/alerts
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Dropped
          </td>
          <td
            className="m-0 row"
          />
        </tr>
      </tbody>
    </table>
  </div>,
]
`;

exports[`Status Status Table should match table snapshot 1`] = `
Array [
  <h2>
    Runtime Information
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Start time
          </td>
          <td
            className="m-0 row"
          >
            Wed, 30 Oct 2019 20:03:23 GMT
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Working directory
          </td>
          <td
            className="m-0 row"
          >
            /home/boyskila/Desktop/prometheus
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Goroutines
          </td>
          <td
            className="m-0 row"
          >
            37
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GOMAXPROCS
          </td>
          <td
            className="m-0 row"
          >
            4
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GOGC
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            GODEBUG
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            WAL corruptions
          </td>
          <td
            className="m-0 row"
          >
            0
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Head chunks
          </td>
          <td
            className="m-0 row"
          >
            1383
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Head time series
          </td>
          <td
            className="m-0 row"
          >
            461
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Last successful configuration reload
          </td>
          <td
            className="m-0 row"
          >
            2019-10-30T22:03:23+02:00
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Configuration reload
          </td>
          <td
            className="m-0 row"
          >
            Successful
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Storage retention
          </td>
          <td
            className="m-0 row"
          >
            15d
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <h2>
    Build Information
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            version
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            revision
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            branch
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            buildUser
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            buildDate
          </td>
          <td
            className="m-0 row"
          >
            
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            goVersion
          </td>
          <td
            className="m-0 row"
          >
            go1.13.3
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <h2>
    Alertmanagers
  </h2>,
  <div
    className="table-responsive"
  >
    <table
      className="h-auto table table-sm table-borderless table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Active
          </td>
          <td
            className="m-0 row"
          >
            <a
              className="mr-5"
              href="https://1.2.3.4:9093/api/v1/alerts"
            >
              https://1.2.3.4:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.5:9093/api/v1/alerts"
            >
              https://1.2.3.5:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.6:9093/api/v1/alerts"
            >
              https://1.2.3.6:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.7:9093/api/v1/alerts"
            >
              https://1.2.3.7:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.8:9093/api/v1/alerts"
            >
              https://1.2.3.8:9093/api/v1/alerts
            </a>
            <a
              className="mr-5"
              href="https://1.2.3.9:9093/api/v1/alerts"
            >
              https://1.2.3.9:9093/api/v1/alerts
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "width": "35%",
              }
            }
          >
            Dropped
          </td>
          <td
            className="m-0 row"
          />
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
